zebROS_ws/src/tf_object_detection/src/pub_video.py

```diff
@@ -14,32 +14,47 @@ def __init__(self):
         rospy.init_node('pub_video', anonymous=False)
         rospy.on_shutdown(self.cleanup)
 
-        self.filename = "video.mp4"
+        self.filename = "testvid.mp4"
         self.pub_topic = "c920/rect_image"
         self.framerate = 30
-        self.show_video = False
+        self.show_video = True
+        #added ~ to all the topics
+        if rospy.has_param('~pub_topic'):
+            self.pub_topic = rospy.get_param('~pub_topic')
 
-        if rospy.has_param('pub_topic'):
-            self.pub_topic = rospy.get_param('pub_topic')
+        if rospy.has_param('~filename'):
+            self.filename = rospy.get_param('~filename')
 
-        if rospy.has_param('filename'):
-            self.filename = rospy.get_param('filename')
+        if rospy.has_param('~framerate'):
+            self.framerate = rospy.get_param('~framerate')
 
-        if rospy.has_param('framerate'):
-            self.framerate = rospy.get_param('framerate')
-
-        if rospy.has_param('show_video'):
-            self.show_video = rospy.get_param('show_video')
+        if rospy.has_param('~show_video'):
+            self.show_video = rospy.get_param('~show_video')
 
         rospack = rospkg.RosPack()
         image_path = rospack.get_path('tf_object_detection') + '/src/'
         self.capture = cv2.VideoCapture(image_path + str(self.filename))
+        #Can be removed, just to check it acutally uses the config
+        print(self.filename, self.framerate, self.pub_topic)
```

Comment:
For these, add a label so we know what they are. The huge log of messages at startup can sometimes get mixed up, so even prefixing them with "Opening video file " + ... would be helpful

Reply:
Ok, should be more descriptive now.